app/controllers/concerns/json_collection_pagination.rb

```diff
@@ -67,4 +58,38 @@ def paginated_json_collection(collection, path_helper, sort_key, sort_order, off
       { json: data_on_page, meta: meta }
     end
   end
+
+  private
+
+  def query_params
+    params.dup()
+  end
+
+  def sort_key
+    (params[:sort_key].presence || :id).to_sym
+  end
+
+  def sort_order
+    params[:sort_order].to_s == 'desc' ? :desc : :asc
+  end
+
+  def sort_reorder(collection)
+    key = sort_key
+    fail ArgumentError.new('Invalid sort_key') unless collection.column_names.include?(key.to_s)
+    collection.reorder(key => sort_order)
+  end
+
+  def sort_offset
+    (params[:offset].presence || 0).to_i
+  end
+
+  def sort_per_page
+    return false if ['false', false, '∞'].include?(params[:per_page])
```

Comment:
Took me a while to trace through how this line of logic is still able to disable `limit`ing the results query. Would it be easy to understand if it were up in `paginated_json_collection_new` or in a differently named method?